### PR TITLE
Auto complete

### PR DESCRIPTION
**Pull request by mhoyer**
Please check contained 83070605aa51c4f63a8a117c4f1f4c2348324b26. Not sure why the `IFS` variable needs to be set there at all and if it breaks anything prefixing it with `local`. I bet you possibly corrupted other auto completion libraries accidentally. See also `-W` parameter for `complete` command: https://www.gnu.org/software/bash/manual/html_node/Programmable-Completion-Builtins.html

### CODE REVIEW

File: rapid-git.sh [hunk 1]
```diff
@@ -660,3 +660,22 @@ function rapid {
...
+    COMPREPLY=($(compgen -W "${cmds}" -- ${cur}))
+    return 0
+  fi
+
+  # fallback completion
+  COMPREPLY=($(compgen -o bashdefault -o default -o nospace -f ${cur}))
+  return 0
+}
+
+complete -F _rapid_complete rapid
```
Comment (agross) [inline rapid-git.sh:750]: `complete` is not available on zsh

Comment (agross) [reply rapid-git.sh:750]: Perhaps you put the whole function definition in a conditional `_rapid_zsh`?

**Review by agross — CHANGES_REQUESTED** [review]

File: rapid-git.sh [hunk 2]
```diff
@@ -660,3 +660,22 @@ function rapid {
...
   return $exit_status
 }
+
+_rapid_complete() 
+{
+  COMPREPLY=()
+  local cur="${COMP_WORDS[COMP_CWORD]}"
+
+  if [[ $COMP_CWORD -eq 1 ]] ; then
+    local cmds="branch checkout diff drop merge push rebase remove stage status track unstage"
```
Comment (agross) [inline rapid-git.sh:740]: DRY, these can be detected. See `__rapid_command_not_found`.

File: rapid-git.sh [hunk 3]
```diff
@@ -39,7 +39,7 @@ function rapid {
 
     if ! __rapid_zsh; then
       # Bash uses declare to return all functions.
-      IFS=$'\n'
+      local IFS=$'\n'
```
Comment (agross) [inline rapid-git.sh]: Should be fixed in master

**Comment by GregOnNet** [comment]
Hey, what is the state of this PR?
Can it be merged? It looks like Marcel added fixes as requested. :-)

**Comment by agross** [comment]
State is the same as 8 years ago. I think both of us didn't look at it any more.

**Pull request closed without merging**